compiler/x/codegen/BinaryEvaluator.cpp

```diff
@@ -1937,13 +1937,59 @@ TR::Register *OMR::X86::TreeEvaluator::integerMulEvaluator(TR::Node *node, TR::C
    TR::Node     *firstChild     = node->getFirstChild();
    TR::Node     *secondChild    = node->getSecondChild();
    bool          nodeIs64Bit    = TR::TreeEvaluator::getNodeIs64Bit(node, cg);
+   bool isMemRef                = firstChild->getOpCode().isMemoryReference();
+   TR_X86OpCodes opCode, regOpCode, memOpCode, movOpCode;
    intptrj_t    value;
 
    if (node->isDualCyclic())
       {
       return TR::TreeEvaluator::integerDualMulEvaluator(node, cg);
       }
 
+   // Opcode evaluation for bmul, smul imul and lmul
+   switch (secondChild->getDataType())
+      {
+      case TR::Int8:
+         value = secondChild->getByte();
+         opCode = isMemRef ? IMUL1AccMem : IMUL1AccReg;
+         regOpCode = IMUL1AccReg;
+         memOpCode = IMUL1AccMem;
+         movOpCode = MOV1RegReg;
+         break;
+
+      case TR::Int16:
+         value = secondChild->getShortInt();
+         if (isMemRef)
+            opCode = (value >= -128 && value <= 127) ? IMUL2RegMemImms : IMUL2RegMemImms;
```

Comment:
I think there is a bug here why are both instructions on the if the same? shouldn't the second one be IMULRegMemImm2? if this is a bug can you please make sure there is a test case to handle this code paths because current tests don't necessarily seem to be handling this...